src/main/kotlin/no/nav/klage/oppgave/service/KlagebehandlingMegaPutFacade.kt

```diff
@@ -0,0 +1,171 @@
+package no.nav.klage.oppgave.service
+
+import no.nav.klage.oppgave.api.view.KlagebehandlingEditableFieldsInput
+import no.nav.klage.oppgave.api.view.TilknyttetDokument
+import no.nav.klage.oppgave.domain.klage.Klagebehandling
+import no.nav.klage.oppgave.domain.kodeverk.Grunn
+import no.nav.klage.oppgave.domain.kodeverk.Hjemmel
+import no.nav.klage.oppgave.domain.kodeverk.Utfall
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.util.*
+
+@Service
+@Transactional
+class KlagebehandlingMegaPutFacade(
+    private val klagebehandlingService: KlagebehandlingService,
+    private val vedtakService: VedtakService
+) {
+
+    fun updateEditableFields(
+        klagebehandlingId: UUID,
+        input: KlagebehandlingEditableFieldsInput,
+        innloggetIdent: String
+    ): Klagebehandling {
+        val klagebehandling =
+            klagebehandlingService.getKlagebehandlingForUpdate(klagebehandlingId, input.klagebehandlingVersjon)
+
+        dirtyCheckAndUpdateInternVurdering(input, klagebehandling, innloggetIdent)
+        dirtyCheckAndUpdateSendTilbakemelding(input, klagebehandling, innloggetIdent)
+        dirtyCheckAndUpdateTilbakemelding(input, klagebehandling, innloggetIdent)
+        dirtyCheckAndUpdateUtfall(input, klagebehandling, innloggetIdent)
+        dirtyCheckAndUpdateGrunn(input, klagebehandling, innloggetIdent)
+        dirtyCheckAndUpdateHjemlerInVedtak(input, klagebehandling, innloggetIdent)
+        dirtyCheckAndUpdateDokumentReferanser(input, klagebehandling, innloggetIdent)
+        return klagebehandling
+    }
+
+    private fun dirtyCheckAndUpdateDokumentReferanser(
+        input: KlagebehandlingEditableFieldsInput,
+        klagebehandling: Klagebehandling,
+        innloggetIdent: String
+    ) {
+        val nyVerdi = input.tilknyttedeDokumenter
+        val gammelVerdi =
+            klagebehandling.saksdokumenter.map { TilknyttetDokument(it.journalpostId, it.dokumentInfoId) }.toSet()
+        if (isDirty(gammelVerdi, nyVerdi)) {
```

Comment:
Tror ikke isDirty() fungerer for disse objektene (grunnet `!=`)

Reply:
Ja, jeg burde legge til en test, enig i det. Jeg skyndtet meg veldig i går.. Men jeg fant denne, så jeg tror det skal fungere: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/-abstract-set/equals.html
(Jeg startet med å kode det mer komplisert..)

Reply:
Har rekkefølge noe å si?

Reply:
Leste ditt svar nå :) Flott. Da er sikkert alt bra.

Reply:
Kotlin løser alt!